### PR TITLE
Feature(GuidedTour)/ Init helper plugin + admin

### PR DESCRIPTION
**Pull request by vincentbpro**
### What does it do?

- adds guided tour context, hook and provider to helper-plugin
- in the admin: using helpers + init reducer

### CODE REVIEW

File: packages/core/admin/admin/src/components/GuidedTour/reducer.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export const initialState = {
+  currentStep: null,
+};
+
+const reducer = (state = initialState, action) =>
+  produce(state, draftState => {
+    switch (action.type) {
+      case 'SET_STEP': {
+        // console.log(action, state, draftState);
```
Comment (soupette) [inline packages/core/admin/admin/src/components/GuidedTour/reducer.js:12]: You can remove the log here.

Also can you add the following :

```js
 draftState.currentStep = action.step;
 break;
```

File: packages/core/admin/admin/src/components/GuidedTour/reducer.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable consistent-return */
+import produce from 'immer';
+
+export const initialState = {
+  currentStep: null,
+};
+
+const reducer = (state = initialState, action) =>
```
Comment (soupette) [inline packages/core/admin/admin/src/components/GuidedTour/reducer.js:8]: Can you add the skeleton of the reducer.test.js as well?

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/core/admin/admin/src/components/GuidedTour/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import PropTypes from 'prop-types';
+import { GuidedTourProvider } from '@strapi/helper-plugin';
+import reducer, { initialState } from './reducer';
+
+const GuidedTour = ({ children }) => {
+  const [{ currentStep }, dispatch] = useReducer(reducer, initialState);
+
+  const setStep = step => {
+    dispatch({
+      type: 'SET_STEP',
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/components/GuidedTour/index.js:19]: Can you rename this action to SET_CURRENT_STEP please 🙏

File: packages/core/admin/admin/src/components/GuidedTour/reducer.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  produce(state, draftState => {
+    switch (action.type) {
+      case 'SET_STEP': {
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/components/GuidedTour/reducer.js:11]: SET_CURRENT_STEP

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

**Review by soupette — APPROVED** [review]
LGTM

**Review by HichamELBSI — APPROVED** [review]
LGTM 👍

**Pull request merged**